### DISCUSSION TITLE
Fast Refresh had to perform a full reload loop

### DISCUSSION THREAD

**Discussion by ricardo-fnd** [vercel/next.js#59580] [Help]
### Summary

I have an application at `localhost:8080` that receives every request and rewrites to the correct app. This application is called **router**.
In the image, you can see the **router** receives the request, and rewrites to the correct app depending on the path. (for example, `localhost:3001/login`)

When rewriting to an application using that is still using `pages` folder, the **router** app "throws" `Fast Refresh had to perform a full reload.` in a loop.
When rewriting to an application using `app` folder, the problem disappears.

Every app is using **NextJS v14.04** but some apps are still using `pages` folder and others are using the new `app` folder.
This only happens in apps using `pages` folder

![Screenshot 2023-12-13 at 13 43 05](https://github.com/vercel/next.js/assets/52207061/48323ccc-5ec6-45b8-9e9c-2ce9c94c4ef9)

**Comment by ricardo-fnd** [comment]
Actually, in those apps that work (`app` folder) **fast refresh** doesn't work, I have to manually press F5.
But if I go directly to `localhost:3001/login` instead of `localhost:8080/login` (that rewrites to **localhost:3001**) fast refresh works. So this is a problem with rewrites?

**Comment by ricardo-fnd** [comment]
https://github.com/vercel/next.js/assets/52207061/bce23dd7-850c-4e72-8afe-40402c265e39

as you can see, we are in `login` app, but for some reason, the `router` app full reloads... but if access `login` app directly, this doesn't happen

**Comment by ricardo-fnd** [comment]
anyone to help? I'm going crazy with this dev experience

**Comment by ricardo-fnd** [comment]
Just noticed https://github.com/vercel/next.js/issues/59161

**Comment by malazaysc** [comment]
can confirm this still happens in the latest canary version
